src/components/user-settings-tabs/InvitesSettings.js

```diff
@@ -32,51 +32,74 @@ const styles = {
   }
 }
 
-const getExpirationDate = (invite) => {
-  const daysUntilExpiration = parseInt(invite.daysFromCreationUntilExpiration, 10)
-  const millisPerDay = 24 * 60 * 60 * 1000
-  const dateCreatedMillis = Date.parse(invite.createdAt)
-  return new Date(dateCreatedMillis + (millisPerDay * daysUntilExpiration))
-}
+class InvitesSettings extends Component {
+  static getExpirationDate(invite) {
+    const daysUntilExpiration = parseInt(invite.daysFromCreationUntilExpiration, 10)
+    const millisPerDay = 24 * 60 * 60 * 1000
+    const dateCreatedMillis = Date.parse(invite.createdAt)
+    return new Date(dateCreatedMillis + (millisPerDay * daysUntilExpiration))
+  }
+
+  static getInviteActions(statusOpen, onAcceptInvite, onDeclineInvite) {
+    if (statusOpen) {
+      return (
+        <span>
+          <RaisedButton label="Accept" primary style={styles.acceptButton} onClick={onAcceptInvite} />
+          <RaisedButton label="Decline" secondary onClick={onDeclineInvite} />
+        </span>
+      )
+    }
+    return <span />
+  }
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      snackOpen: false
+    }
+    this.getTableRow = this.getTableRow.bind(this)
+    this.closeSnackbar = this.closeSnackbar.bind(this)
+    this.acceptInviteHandler = this.acceptInviteHandler.bind(this)
+    this.declineInviteHandler = this.declineInviteHandler.bind(this)
+  }
 
-const getInviteActions = (statusOpen, onAcceptInvite, onDeclineInvite) => {
-  if (statusOpen) {
+  getTableRow(invite) {
+    const statusOpen = (invite.status.toLowerCase() === 'open')
+    const rowStyles = (statusOpen) ? {} : styles.rowClosed
+    const inviteActions = InvitesSettings.getInviteActions(
+      statusOpen,
+      this.acceptInviteHandler(invite.id),
+      this.declineInviteHandler(invite.id)
+    )
+    const expirationDate = InvitesSettings.getExpirationDate(invite)
     return (
-      <span>
-        <RaisedButton label="Accept" primary style={styles.acceptButton} onClick={onAcceptInvite} />
-        <RaisedButton label="Decline" secondary onClick={onDeclineInvite} />
-      </span>
+      <TableRow key={invite.id}>
+        <TableRowColumn style={rowStyles}>{invite.projectName}</TableRowColumn>
+        <TableRowColumn style={rowStyles}>{expirationDate.toDateString()}</TableRowColumn>
+        <TableRowColumn style={rowStyles}>{invite.status.toUpperCase()}</TableRowColumn>
+        <TableRowColumn style={rowStyles}>{inviteActions}</TableRowColumn>
+      </TableRow>
     )
   }
-  return <span />
-}
 
-const getTableRow = (invite) => {
-  function acceptInvite() {
-    // TODO
-    console.log(`TODO: Accepting invite ${invite.id}...`)
+  acceptInviteHandler(inviteId) {
+    return async () => {
```

Comment:
Why are you doing this for these functions? Why not just do
```
async acceptInvite(inviteId) {
    await .....
}
```

Reply:
It encapsulates the `inviteId` in a closure. The actual caller of the returned function is an `onClick` event, and it doesn't know the `inviteId`.

Reply:
Actually nvm. This is good!